registry/client/auth/session.go

```diff
@@ -209,7 +210,7 @@ func (th *tokenHandler) fetchToken(params map[string]string) (token string, err
 	}
 	defer resp.Body.Close()
 
-	if resp.StatusCode != http.StatusOK {
+	if !client.SuccessStatus(resp.StatusCode) {
```

Comment:
I agree. At some point, we may decide the API is completely stable and will lock down all status codes forever.